respect-server/src/main/kotlin/world/respect/server/routes/school/respect/schoolRegistrationWebRoute.kt

```diff
@@ -0,0 +1,222 @@
+                            +"School Name"
+                        }
+                        input(type = InputType.text, name = "schoolName") {
+                            id = "schoolName"
+                            required = true
+                            placeholder = "Enter school name"
+                        }
+                    }
+
+                    when (schoolConfig.registration.mode) {
+                        SchoolConfig.RegistrationConfig.RegistrationMode.SUBDOMAIN -> {
+                            div("form-group") {
+                                label {
+                                    htmlFor = "schoolSubdomain"
+                                    +"School Link"
+                                }
+                                input(type = InputType.text, name = "schoolSubdomain") {
+                                    id = "schoolSubdomain"
+                                    required = true
+                                    placeholder = "school-name"
+                                    onInput = "updateFullUrl()"
+                                }
+                            }
+                            // Hidden field for full URL
+                            input(type = InputType.hidden, name = "schoolUrl") {
+                                id = "schoolUrl"
+                                value =
+                                    "https://school-name.${schoolConfig.registration.topLevelDomain}"
+                            }
+                        }
+
+                        SchoolConfig.RegistrationConfig.RegistrationMode.ANY_URL -> {
+                            div("form-group") {
+                                label {
+                                    htmlFor = "schoolUrl"
+                                    +"School URL"
+                                }
+                                input(type = InputType.text, name = "schoolUrl") {
+                                    id = "schoolUrl"
+                                    required = true
+                                }
+                            }
+                        }
+
+                        else -> {
+                            // Should not reach here since registration is disabled
+                        }
+                    }
+
+                    // Hidden redirect field for deep linking to app
+                    input(type = InputType.hidden, name = "redirect") {
+                        value = call.request.queryParameters["redirect"]
+                            ?: "world.respect.app://school-registered"
+                    }
+
+                    button(type = ButtonType.submit, classes = "submit-btn") {
+                        +"Next"
+                    }
+                }
+            }
+        }
+    }
+
+// Handle form submission
+    post("/register-school") {
+        // Check if registration is enabled
+        if (!schoolConfig.registration.enabled) {
```

Comment:
The only code should be here in this route is the code that will get the required input parameters so that it can invoke the use case.